### PR TITLE
Generic

### PR DESCRIPTION
**Pull request by KuroPSPiso**

### CODE REVIEW

File: src/main/java/com/molvenolakeresort/security/WebSecurityConfiguration.java [hunk 1]
```diff
@@ -23,12 +25,16 @@ public void configure(AuthenticationManagerBuilder auth) throws Exception {
...
-
-        //Login,logout page and resources are permitted for all users
-        http.authorizeRequests().antMatchers("/", "/login", "/logout", "/resources/**").permitAll();
+        http.authorizeRequests().anyRequest().authenticated()
+                .and()
+                .httpBasic();
+//        //CSFR is disabled,if you dont know what csrf is,Spring has a beautiful documentaion about it ,Check it out.
+//        http.csrf().disable();
+//        //Without the following line(s) the h2 database will return a blank page.
+//        http.headers().frameOptions().disable();
```
[the remark this answers is not in the document]
Comment (KuroPSPiso) [reply src/main/java/com/molvenolakeresort/security/WebSecurityConfiguration.java]: Required at a later date

File: src/main/java/com/molvenolakeresort/controllers/generic/GuestController.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
+package com.molvenolakeresort.controllers.generic;
+
+import com.molvenolakeresort.controllers.exceptions.EntityNotFoundException;
+import com.molvenolakeresort.models.generic.security.GuestInformation;
+import com.molvenolakeresort.models.generic.security.Profile;
+import com.molvenolakeresort.services.GenericService;
+import com.molvenolakeresort.services.UserService;
+import com.molvenolakeresort.services.impl.GenericServiceImpl;
+import com.molvenolakeresort.services.impl.UserServiceImpl;
```
[the remark this answers is not in the document]
Comment (KuroPSPiso) [reply src/main/java/com/molvenolakeresort/controllers/generic/GuestController.java]: to be used

File: src/main/java/com/molvenolakeresort/security/WebSecurityConfiguration.java [hunk 3]
```diff
@@ -23,12 +25,16 @@ public void configure(AuthenticationManagerBuilder auth) throws Exception {
...
+//
+//        //Login,logout page and resources are permitted for all users
+//        http.authorizeRequests().antMatchers("/", "/login", "/logout", "/resources/**").permitAll();
```
[the remark this answers is not in the document]
Comment (KuroPSPiso) [reply src/main/java/com/molvenolakeresort/security/WebSecurityConfiguration.java]: to be reenabled on later development

File: src/main/java/com/molvenolakeresort/services/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -18,113 +21,133 @@
...
+        } else if (eventVisitor != null) {
+            template = eventVisitor;
+        } else if (restaurantVisitor != null) {
+            template = restaurantVisitor;
+        }
+
+        //if the guest does not yet exist as a visitor create one.
+        if (template == null) {
+            template = createProfileTemplate(guest, false);
+            if (template != null) {
```
[the remark this answers is not in the document]
Comment (KuroPSPiso) [reply src/main/java/com/molvenolakeresort/services/impl/UserServiceImpl.java:129]: False Positive

**Review by erwinjacobs90 — APPROVED** [review]

**Review by LucasWalters — COMMENTED** [review]
Looks great

**Review by GaryMikoen — APPROVED** [review]

**Review by LucasWalters — APPROVED** [review]

**Pull request merged**